File: tools/benchmarks/merlo/benchmarks/stage06p_extended/corpus/interface_dispatch/python/main.py
import sys
MASK=(1<<64)-1
class Square:
    __slots__=()
    def apply(self,value): return value*value
class Increment:
    __slots__=()
    def apply(self,value): return value+1
def run(n):
    operations=(Square(),Increment());total=0
    for i in range(n): total=(total+operations[i&1].apply(i))&MASK
    return total
print(run(int(sys.argv[1])))

File: tools/benchmarks/merlo/benchmarks/stage06p_extended/corpus/interface_dispatch/rust/main.rs
trait Operation{fn apply(&self,value:u64)->u64;}struct Square;struct Increment;impl Operation for Square{fn apply(&self,value:u64)->u64{value.wrapping_mul(value)}}impl Operation for Increment{fn apply(&self,value:u64)->u64{value.wrapping_add(1)}}
fn run(n:u64)->u64{let square=Square;let increment=Increment;let operations:[&dyn Operation;2]=[&square,&increment];let mut sum=0u64;for i in 0..n{sum=sum.wrapping_add(operations[(i&1)as usize].apply(i));}sum}
fn main(){let n=std::env::args().nth(1).unwrap().parse::<u64>().unwrap();println!("{}",run(n));}
